app/controllers/articles_controller.rb

```diff
@@ -15,8 +15,8 @@ def new
 
   def create
     p = article_params
-    to_save = {title: p[:title], body: p[:body], user_id: current_user.id}
-    @article = Article.create(to_save)
+    p[:user_id] = current_user.id
+    @article = Article.create(p)
```

Comment:
Consider using `u.articles.create(...)` instead, that way you avoid the problem of the missing `user_id`.